components/admin_console/billing/billing_history.tsx

```diff
@@ -94,62 +63,90 @@ const getPaymentStatus = (status: string) => {
                 />
             </div>
         );
-    case 'Pending':
+    case 'paid':
         return (
-            <div className='BillingHistory__paymentStatus pending'>
+            <div className='BillingHistory__paymentStatus paid'>
                 <i className='icon icon-check-circle-outline'/>
                 <FormattedMessage
-                    id='admin.billing.history.pending'
-                    defaultMessage='Pending'
+                    id='admin.billing.history.paid'
+                    defaultMessage='Paid'
                 />
             </div>
         );
-    case 'Paid':
+    default:
         return (
-            <div className='BillingHistory__paymentStatus paid'>
+            <div className='BillingHistory__paymentStatus pending'>
                 <i className='icon icon-check-circle-outline'/>
                 <FormattedMessage
-                    id='admin.billing.history.paid'
-                    defaultMessage='Paid'
+                    id='admin.billing.history.pending'
+                    defaultMessage='Pending'
                 />
             </div>
         );
-    default:
-        return null;
     }
 };
 
 const BillingHistory: React.FC<Props> = () => {
-    //const [billingHistory, setBillingHistory] = useState(billingInfo);
-    // TO-DO: Wire this up with real data
-    let billingHistory: undefined | Array<any>;
-    const previousPage = () => {};
-    const nextPage = () => {};
+    const dispatch = useDispatch();
+    const invoices = useSelector((state: GlobalState) => state.entities.cloud.invoices);
+    const subscription = useSelector((state: GlobalState) => state.entities.cloud.subscription);
+    const product = useSelector((state: GlobalState) => {
+        if (state.entities.cloud.products && subscription) {
+            return state.entities.cloud.products[subscription?.product_id];
+        }
+        return undefined;
+    });
+
+    const [billingHistory, setBillingHistory] = useState<Invoice[] | undefined>(undefined);
+    const [firstRecord, setFirstRecord] = useState(1);
+
+    const previousPage = () => {
+        if (firstRecord > PAGE_LENGTH) {
+            setFirstRecord(firstRecord - PAGE_LENGTH);
+        }
+    };
+    const nextPage = () => {
+        if (invoices && (firstRecord + PAGE_LENGTH) < Object.values(invoices).length) {
+            setFirstRecord(firstRecord + PAGE_LENGTH);
+        }
+
+        // TODO: When server paging, check if there are more invoices
+    };
 
     useEffect(() => {
+        dispatch(getCloudProducts());
+        dispatch(getCloudSubscription());
+        dispatch(getInvoices());
         pageVisited('cloud_admin', 'pageview_billing_history');
     }, []);
 
+    useEffect(() => {
```

Comment:
Why use `useEffect` two times?

Reply:
First `useEffect` only runs once as there are no dependent parameters (see the second argument), so it acts like `componentDidMount`.

Second `useEffect` updates other state when the two dependent parameters change.

Reply:
Cool, good to know :)